### PR TITLE
ZEL-1772 Implement Public Keys API

### PR DESCRIPTION
**Pull request by vm-vchyypesh**
- added endpoint `/.well-known/jwks.json`
- added JWK service with rotating keys every 24 hours
- updated Node.js docker image to latest 18 version

### CODE REVIEW

**Review by vm-vhryschenko — DISMISSED** [review]

**Review by K-Mistele — DISMISSED** [review]

File: src/services/jwk.service.ts [hunk 1]
```diff
@@ -45,6 +46,21 @@ export class JWKService {
...
+            .then(jwkService => {
+              return jwkService.rotateKeys();
+            })
+            .then(() => logService.log('JWKService rotated keys!'))
+            .catch((err: unknown) => {
+              logService.log('An error occurred while the JWKService was rotating keys!');
+              logService.error(err);
+            }),
+        KEY_ROTATION_INTERVAL
+      );
```
Comment (K-Mistele) [inline src/services/jwk.service.ts]: It occurs to me that I should check how evervault handles this - if enclaves are transparently horizontally scaled and load-balanced, this approach will cause problems.

Comment (K-Mistele) [reply src/services/jwk.service.ts]: I have a meeting scheduled with Evervault in the morning to discuss.

Comment (vm-vchyypesh) [reply src/services/jwk.service.ts]: @K-Mistele An additional question: Could you please clarify where the keys should be stored?

Comment (K-Mistele) [reply src/services/jwk.service.ts]: Yes, after my meeting with evervault I should have a better idea of how to approach this )

**Comment by K-Mistele** [comment]
Hi guys - here is what I learned from Evervault:
1. Enclaves _are_ horizontally scaled; there are two at a time
2. The load balancer does _not_ have "sticky sessions" enabled, meaning it is _not_ guaranteed that a user will always be connecting to the same enclave. 

This means that we need to figure out a solution to key generation & distribution in the enclave to ensure that both have the same key. I will do some thinking & design on this today, and hopefully we can tackle it next week.

**Review by vm-vhryschenko — DISMISSED** [review]

**Comment by vm-vchyypesh** [comment]
@K-Mistele @aleksei-yudin 
AWS S3 is used for external key storage, where the keys are encrypted using the Evervault SDK. The BUCKET_NAME and S3_KEY can be configured through environment variables, also there are default values available for tests and local development. To enable proper key rotation, a pseudo-lock mechanism has been implemented using an AWS S3 object in the same bucket.
We need to create separate buckets for each environment, set up a user for Evervault Enclaves with access to these buckets, provide credentials (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY) to the Enclaves environments, and establish a connection between S3 and the Enclaves.

**Review by aleksei-yudin — DISMISSED** [review]

**Review by vm-vhryschenko — DISMISSED** [review]

**Review by vm-vhryschenko — DISMISSED** [review]

**Review by vm-vhryschenko — DISMISSED** [review]

**Review by aleksei-yudin — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -15,24 +18,27 @@
     "prettier": "prettier --write 'src/**/*.ts' 'tests/**/*.ts'"
   },
   "dependencies": {
+    "@aws-sdk/client-s3": "^3.631.0",
     "@evervault/sdk": "^4.3.0",
-    "axios": "^1.4.0",
-    "body-parser": "^1.20.2",
+    "axios": "^1.7.3",
```
Comment (K-Mistele) [inline package.json]: Is this the version of axios that was patched against the SSRF vulnerability?

**Review by K-Mistele — DISMISSED** [review]
LGTM

File: package.json (hunk 2 above)
Comment (vm-vchyypesh) [reply package.json]: `npm audit` doesn't show high or critical vulnerabilities for this version, however I will update to latest version

Comment (vm-vchyypesh) [reply package.json]: `axios` has been updated to version `1.7.7`

**Review by K-Mistele — APPROVED** [review]
LGTM ))

**Pull request merged**